server/src/main/java/org/elasticsearch/common/lucene/Lucene.java

```diff
@@ -384,6 +384,10 @@ public static ScoreDoc readScoreDoc(StreamInput in) throws IOException {
         return new ScoreDoc(in.readVInt(), in.readFloat());
     }
 
+    public static ScoreDoc readScoreDocWithShardIndex(StreamInput in) throws IOException {
+        return new ScoreDoc(in.readVInt(), in.readFloat(), in.readInt());
```

Comment:
Nit: the shard index should be greater than or equal to -1, so we could read it with `in.readVInt() - 1` to hopefully always encode it on a single byte.

Reply:
👍